### ISSUE TITLE
non-readable files in source directory massively stall backup ("borg create")

### ISSUE THREAD

**Issue by agrommek** [borgbackup/borg#7446]
## Have you checked borgbackup docs, FAQ, and open GitHub issues?

yes

## Is this a BUG / ISSUE report or a QUESTION?

Issue / Feature Request / bug?

#### Your borg version (borg -V).

borg-2.0.0b5

#### Operating system (distribution) and version.

Ubuntu Server 22.04
#### Hardware / network configuration, and filesystems used.

Intel(R) Core(TM) i3-8145U CPU @ 2.10GHz
16 GB RAM
ext4

#### How much data is handled by borg?

Not much, just test data. Maybe 5 MB.

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

`/path/to/borg-2.0.0b5-linuxnew64 create --stats --list --checkpoint-interval=60 --repo /path/to/testrepo1 --verbose archive-{utcnow} /etc`

## Describe the problem you're observing.

I am getting my feet wet with the new syntax and related repositories coming in borg 2.0. For that purpose I created some test repos and backed up my /etc directory (as a dummy source directory - a directory with many text files which do not change while running borg). Some files under this directory are not readable for the user borg runs under. This causes borg to log a warning and retry for a total of about 15 seconds - for each file. Example:

```
/etc/sudoers.d/README: open: [Errno 13] Permission denied: 'README', slept 0.001s, next: retry: 1 of 9...
/etc/sudoers.d/README: open: [Errno 13] Permission denied: 'README', slept 0.003s, next: retry: 2 of 9...
/etc/sudoers.d/README: open: [Errno 13] Permission denied: 'README', slept 0.010s, next: retry: 3 of 9...
/etc/sudoers.d/README: open: [Errno 13] Permission denied: 'README', slept 0.032s, next: retry: 4 of 9...
/etc/sudoers.d/README: open: [Errno 13] Permission denied: 'README', slept 0.100s, next: retry: 5 of 9...
/etc/sudoers.d/README: open: [Errno 13] Permission denied: 'README', slept 0.316s, next: retry: 6 of 9...
/etc/sudoers.d/README: open: [Errno 13] Permission denied: 'README', slept 1.000s, next: retry: 7 of 9...
/etc/sudoers.d/README: open: [Errno 13] Permission denied: 'README', slept 3.162s, next: retry: 8 of 9...
/etc/sudoers.d/README: open: [Errno 13] Permission denied: 'README', slept 10.000s, next: retry: 9 of 9...
/etc/sudoers.d/README: open: [Errno 13] Permission denied: 'README'
E /etc/sudoers.d/README
```
While waiting for the file(s) to become accessible (which never happens, of course) the backup stalls. This causes the backup to run for about 14 minutes for a measly 4.6 MB of source files. For comparison: Running with the correct permissions (i.e. all files are readable by borg) the backup take WAY less than one second.

Suggestion / feature request:

Either make waiting for "Permission denied" errors non-blocking by using asyncio or threads (--> backup can continue while borg waits) and/or provide an option to skip non-accessible files. 

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

yes - does not appear to be a bug but rather "works as intended". But maybe not very smart... ;-)

#### Include any warning/errors/backtraces from the system logs

None

**Comment by ThomasWaldmann** [comment]
Oh. Yeah, guess we should not retry on "Permission denied", because that won't go away by retrying.

**Comment by ThomasWaldmann** [comment]
Will be fixed by #7452.
